### PR TITLE
feat: Skia lifecycle changes

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

Replaces #12354

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 777c7d6</samp>

This pull request improves the support for the Skia backend on GTK and WPF platforms, by refactoring and simplifying some classes, using the new `GtkHost.Current.MainWindow` property, adding a new `GtkApplicationViewExtension` class, and registering the GTK-specific extensions. It also adapts the `SamplesApp.Skia.WPF` project to use the Skia host and removes some unused files.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jeromelaban** [comment]
@MartinZikmund
```
##[error]EXEC(0,0): Error : Removed type Uno.UI.Skia.Platform.WpfHost not found in ignore set.
```

File: build/PackageDiffIgnore.xml [hunk 1]
```diff
@@ -839,6 +839,10 @@
 			<!-- BEGIN Skia Adjustments -->
 			<Member fullName="SkiaSharp.SKSize Uno.UI.Skia.Platform.WpfHost::CanvasSize()" reason="Incorrect type" />
 			<!-- END Skia Adjustments -->
+			
+			<Member fullName="System.Boolean Uno.UI.Skia.Platform.WpfHost.get_IsIsland()" reason="Not needed" />
+			<Member fullName="Windows.UI.Xaml.UIElement Uno.UI.Skia.Platform.WpfHost.get_RootElement()" reason="Not needed" />
+			<Member fullName="Uno.UI.Skia.Platform.WpfHost Uno.UI.Skia.Platform.WpfHost.get_Current()" reason="Not needed" />
```
Comment (jeromelaban) [inline build/PackageDiffIgnore.xml:845]: Let's make it easier for the upcoming 5x merge :)

```suggestion
<!-- BEGIN Skia Lifecycle -->
			<Member fullName="System.Boolean Uno.UI.Skia.Platform.WpfHost.get_IsIsland()" reason="Not needed" />
			<Member fullName="Windows.UI.Xaml.UIElement Uno.UI.Skia.Platform.WpfHost.get_RootElement()" reason="Not needed" />
			<Member fullName="Uno.UI.Skia.Platform.WpfHost Uno.UI.Skia.Platform.WpfHost.get_Current()" reason="Not needed" />	
<!-- END Skia Lifecycle -->
```

File: src/Uno.UI.Runtime.Skia.Gtk/Extensions/GtkApplicationViewExtension.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		private readonly ApplicationView _owner;
+
+		public GtkApplicationViewExtension(object owner)
+		{
+			_owner = (ApplicationView)owner;
+		}
+
+		public void ExitFullScreenMode()
+		{
+			GtkHost.Current!.MainWindow!.Unfullscreen();
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Gtk/Extensions/GtkApplicationViewExtension.cs:18]: I'm suggesting this, but I'd rather see that we log that the main window is not available.

File: src/Uno.UI.Runtime.Skia.Gtk/Extensions/GtkDisplayInformationExtension.cs [hunk 3]
```diff
@@ -10,27 +10,32 @@ namespace Uno.UI.Runtime.Skia;
...
+		GtkHost.Current!.MainWindowShown += GtkDisplayInformationExtension_MainWindowShown;
+		_dpiHelper = new DpiHelper();
+	}
+
+	private void GtkDisplayInformationExtension_MainWindowShown(object? sender, EventArgs e)
+	{
 		_dpiHelper.DpiChanged += OnDpiChanged;
 	}
 
+	private Gtk.Window GetWindow() => GtkHost.Current!.MainWindow!;
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Gtk/Extensions/GtkDisplayInformationExtension.cs:29]: Let's fail properly if the window is not available.

File: src/Uno.UI.Runtime.Skia.Wpf/Extensions/WpfDragDropExtension.cs [hunk 4]
```diff
@@ -31,11 +34,20 @@ internal class WpfDragDropExtension : IDragDropExtension
 		private readonly long _fakePointerId = Pointer.CreateUniqueIdForUnknownPointer();
 		private readonly DragDropManager _manager;
 
+		private static WpfControl _rootControl;
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Wpf/Extensions/WpfDragDropExtension.cs:37]: Could we make this nullable?

File: src/Uno.UI.Composition/Composition/Compositor.skia.cs [hunk 5]
```diff
@@ -167,7 +169,7 @@ private static void ApplyClip(SKSurface surface, Visual visual)
 			if (!_isDirty)
 			{
 				_isDirty = true;
-				CoreWindow.QueueInvalidateRender();
+				CoreApplication.QueueInvalidateRender();
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI.Composition/Composition/Compositor.skia.cs:172]: Yes, I will definitely be adjusting this in the multi-window PR, this is just temporary 👍

[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI.Composition/Composition/Compositor.skia.cs:172]: Adding a note for myself

**Comment by FrozDark** [comment]
If I understand correctly this will allow us to create multiwindow on skia heads?

**Comment by MartinZikmund** [comment]
> If I understand correctly this will allow us to create multiwindow on skia heads?

@FrozDark Not yet, but the follow up PR might 🙃

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**